ui/test/nuxeo-favorites-toggle-button.test.js

```diff
@@ -48,16 +58,31 @@ suite('nuxeo-favorites-toggle-button', () => {
       ]);
     });
 
+    teardown(() => {
+      if (window.confirm.restore) {
+        window.confirm.restore();
+      }
+    });
+
     test('it should display the document as favorite', () => {
       expect(element.favorite).to.be.true;
     });
 
-    test('toggle should remove document from favorites', async () => {
+    test('toggle should remove document from favorites when clicked ok', async () => {
       // Remove document from favorites by toggling
+
+      sinon.stub(window, 'confirm').returns(true);
       tap(element);
       await waitChanged(element, 'favorite');
       expect(element.favorite).to.be.false;
     });
+
+    test('toggle should not remove document from favorites when clicked Cancel', async () => {
+      sinon.spy(button.$.opRemove, 'execute');
+      sinon.stub(window, 'confirm').returns(false);
+      expect(button.$.opRemove.execute.notCalled).to.be.true;
+      expect(element.favorite).to.be.true;
```

Comment:
The problem with this test remains. We did not interact with the element (i.e., tap/click on it), so we are not actually testing what happens when the user hits the cancel button.